lib/SPIRVProducerPass.cpp

```diff
@@ -6192,6 +6225,35 @@ bool SPIRVProducerPassImpl::selectFromSameObject(Instruction *inst) {
   return false;
 }
 
+bool SPIRVProducerPassImpl::isPointerUniform(Value *ptr) {
+  if (!ptr->getType()->isPointerTy()) {
+    return true;
+  }
+  if (GetStorageClass(ptr->getType()->getPointerAddressSpace()) !=
+      spv::StorageClassStorageBuffer) {
+    return true;
+  }
+
+  static std::set<Value *> nonUniformPointer;
```

Comment:
I didn't notice this before, but I'd prefer not using static here. Instead just move this into the SPIRVProducerPassImpl and clear at the beginning of the pass.

Reply:
Done.
Just out of curiosity, why do you prefer not to use static variables in this case?

Reply:
I think it makes it harder to understand the state of the overall pass.